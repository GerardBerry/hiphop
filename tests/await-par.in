A;
;
B;
A;
;
;
B;
A;
!reset;
;
A;
B;
;

