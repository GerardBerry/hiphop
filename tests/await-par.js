"use hopscript"

var rjs = require("../lib/reactive-js.js");

var sigA = new rjs.Signal("A");
var sigB = new rjs.Signal("B");
var sigO = new rjs.Signal("O");

var prg = <rjs.reactivemachine debug name="awaitpar">
  <rjs.inputsignal ref=${sigA}/>
  <rjs.inputsignal ref=${sigB}/>
  <rjs.outputsignal ref=${sigO}/>
  <rjs.sequence>
    <rjs.parallel>
      <rjs.await signal_name="A"/>
      <rjs.await signal_name="B"/>
    </rjs.parallel>
    <rjs.emit signal_name="O"/>
  </rjs.sequence>
</rjs.reactivemachine>;

exports.prg = prg;
