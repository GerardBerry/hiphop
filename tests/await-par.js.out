await-par> A;
--- Output:
await-par> ;
--- Output:
await-par> B;
--- Output:
await-par> A;
--- Output: O
await-par> ;
--- Output:
await-par> ;
--- Output:
await-par> B;
--- Output:
await-par> A;
--- Output:
await-par> ;
--- Output:
await-par> A;
--- Output:
await-par> B;
--- Output: O
await-par> ;
--- Output:
